lib/compiler-finder.ts

```diff
@@ -258,6 +259,11 @@ export class CompilerFinder {
         })();
         const exe = props('exe', compilerId);
         const exePath = path.dirname(exe);
+        const instructionSet = props<string | number>('instructionSet', '').toString() as InstructionSet | '';
+        assert(
+            instructionSet === '' || InstructionSetsList.includes(instructionSet),
+            `Unexpected instruction set ${instructionSet} ${compilerId}`,
+        );
```

Comment:
Making an attempt to verify the type at runtime here. There's a chance this could become annoying later.